### PR TITLE
Feature 4 - Filter Search

### PR DESCRIPTION
**Pull request by yuanjiat**
## Description
Add the Filter Search feature which related to Drink class. Using the JTextfield to create a text box on the customer main page and allow customers searching drinks by one keyword of drink. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

**Review by lisa272 — APPROVED** [review]
overall good work for adding the filter feature

**Review by yyuan-qin — APPROVED** [review]

**Review by AdwinTazement — APPROVED** [review]
successfully adding the filter search feature of the program with clear logic

File: src/main/java/filtersearch/MyFrame.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        root.add(search_box);
+        search_box.setText("Input the keywords of drink");
+        JButton button = new JButton("OK");
+        root.add(button);
+        button.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                Add_to_Filtered(search_box.getText());
+
```
Comment (lisa272) [inline src/main/java/filtersearch/MyFrame.java:32]: For the backend of the filter search part, you successfully present the base interface of a textbox that lets the customer to input search attributes, but I think you did not relate it to the front end. For example, how to present the filtered drink according to the input of the customer.

**Review by AdwinTazement — COMMENTED** [review]
For this part, you succecfully made Add_to_Filtered return the drinks with the keyword in thier names and discriptions. You can also add the ingredient part such that the customers can find drinks by ingredient, you can also change the input from one keyword to more keywords for more precise searching.

**Pull request merged**